src/charm.py

```diff
@@ -280,11 +315,20 @@ def _create_layer_config(self):
         }
         return layer_config
 
-    def _should_run_setup(self, current_plan, s3info: S3Info):
-        # Return true if no services are planned yet (first run)
-        return not current_plan.services or (
+    def _should_run_setup(self, current_plan: Dict, s3info: Optional[S3Info]) -> bool:
+        """Check if the setup script is to be run.
```

Comment:
Perhaps elaborate a bit on the conditions that should be met ?

Reply:
how about now?